apiserver/facade/watcherregistry.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package facade
+
+import "github.com/juju/worker/v3"
+
+// WatcherRegistry holds all the watchers for a connection.
+// It allows the registration of watchers that will be cleaned up when a
+// connection terminates.
+type WatcherRegistry interface {
```

Comment:
Technically it's a worker registry right? Watcher being the specific.

Reply:
I wanted to be specific so people feel bad about shoving random workers into the registry. For example, the `pingTimeout` and `crossModelOfferURL` conform to a worker, but aren't actually workers. They don't have a lifecycle in the true sense, they're just inturned into a bucket and expected to be around at some point.